### PR TITLE
API Endpoint for Video

### LINKED ISSUE #41: API Endpoint for Video (Backend)
Create four new API endpoints in backend:
- Endpoint for fetching all video (using GET http method)
- Endpoint for creating new video (using POST http method)
- Endpoint for updating the video (using PUT http method)
- Endpoint for deleting the video (using DELETE http method)

Notes:
- you can use existing endpoint for reference
- if in your endpoint there is request body required, always make sure you validate it in controller before passing request body value to the service
- use appropriate response util

### PR DESCRIPTION
**Pull request by GlennPoh**
I am wondering if we embed the video when uploaded so the validation before storing into db won't be necessary?
Closes #41

### CODE REVIEW

File: packages/backend/src/services/video.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+/**
+ * Get all videos
+ * @returns array of video document
+ */
+export const getAllVideos = async (
+  lastId: string,
+  limit: number
+): Promise<VideoDoc[]> => {
+  if (lastId === 'NULL') {
+    return await VideoModel.find().limit(limit)
```
Comment (nyxordinal) [inline packages/backend/src/services/video.ts]: this is a `redundant async` i think
```suggestion
    return VideoModel.find().limit(limit).exec()
```

File: packages/backend/src/services/video.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  }
+  return await VideoModel.find({ _id: { $gt: lastId } }).limit(limit)
```
Comment (nyxordinal) [inline packages/backend/src/services/video.ts]:
```suggestion
  return VideoModel.find({ _id: { $gt: lastId } }).limit(limit).exec()
```

File: packages/backend/src/services/video.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    creator,
+    videoEmbedUrl,
+    isVerified,
+  }
+
+  const options = {
+    new: true,
+  }
+
+  const video = await VideoModel.findOneAndUpdate(conditions, data, options)
```
Comment (nyxordinal) [inline packages/backend/src/services/video.ts]: could you please add condition when the document with given `_id`  not found ?

**Comment by nyxordinal** [comment]
> I am wondering if we embed the video when uploaded so the validation before storing into db won't be necessary?

what you mean ?

**Comment by GlennPoh** [comment]
I mean the stored url format will be the normal one 'www' or the embedded one?

File: packages/backend/src/services/video.ts (hunk 3 above)
Comment (GlennPoh) [reply packages/backend/src/services/video.ts]: insert if it not exist?

Comment (nyxordinal) [reply packages/backend/src/services/video.ts]: no, throwing new error might be better

**Comment by nyxordinal** [comment]
I think it's the normal 'www' one

**Comment by GlennPoh** [comment]
what is the isVerified field used for btw?

**Comment by nyxordinal** [comment]
it is to mark content that is "safe" to display, especially to filter out spam and inappropriate content or content that we want to hide for some reason

**Comment by GlennPoh** [comment]
So we store the video when uploaded and only update the verified field in a later date?

**Comment by nyxordinal** [comment]
> So we store the video when uploaded and only update the verified field in a later date?

yeah, that will be the case

File: packages/backend/src/services/video.ts [hunk 4]
```diff
@@ -64,9 +65,16 @@ export const updateVideo = async (
...
 
-  const video = await VideoModel.findOneAndUpdate(conditions, data, options)
-
-  if (!video) throw new TypeError('Video not found')
+  const video = await VideoModel.findOneAndUpdate(
+    conditions,
+    data,
+    options,
+    (err, res) => {
+      console.log(err)
```
Comment (nyxordinal) [inline packages/backend/src/services/video.ts]: please use the `@logger` instead of console.log()

**Comment by nyxordinal** [comment]
looks good for me

**Comment by nyxordinal** [comment]
as always thanks for contributing 👍 @GlennPoh

**Review by nyxordinal — APPROVED** [review]

**Pull request merged**